src/app/util/attribute-storage.cpp

```diff
@@ -202,12 +223,27 @@ EmberAfStatus emberAfSetDynamicEndpoint(uint16_t index, EndpointId id, EmberAfEn
     emAfEndpoints[index].deviceId      = deviceId;
     emAfEndpoints[index].deviceVersion = deviceVersion;
     emAfEndpoints[index].endpointType  = ep;
+    emAfEndpoints[index].dataVersions  = dataVersionStorage;
     emAfEndpoints[index].networkIndex  = 0;
     // Start the endpoint off as disabled.
     emAfEndpoints[index].bitmask = EMBER_AF_ENDPOINT_DISABLED;
 
     emberAfSetDynamicEndpointCount(MAX_ENDPOINT_COUNT - FIXED_ENDPOINT_COUNT);
 
+    // Initialize the data versions, if we have space.
+    if (dataVersionStorage)
```

Comment:
It is very odd to me that we have this raw pointer to what amounts to a flat array, with no associated size, and getting the size requires calling the correct emberAfClusterCountByIndex, and make sure it matches. Why not use a Span<DataVersion> instead? Then you could also do a size safety check that .size() is >= `emberAfClusterCountByIndex` in validation.

Reply:
Good idea, will do.